src/NHibernate/Mapping/Constraint.cs

```diff
@@ -101,10 +101,10 @@ public virtual string SqlDropString(Dialect.Dialect dialect, string defaultCatal
 		{
 			if (IsGenerated(dialect))
 			{
-				string ifExists = dialect.GetIfExistsDropConstraint(Table, Name);
-				string drop =
-					string.Format("alter table {0} drop constraint {1}", Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema), Name);
-				string end = dialect.GetIfExistsDropConstraintEnd(Table, Name);
+				var tableName = Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema);
+				string ifExists = dialect.GetIfExistsDropConstraint(Table, Name, defaultCatalog, defaultSchema);
+				string drop = string.Format("alter table {0} drop constraint {1}", Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema), Name);
+				string end = dialect.GetIfExistsDropConstraintEnd(Table, tableName, defaultCatalog, defaultSchema);
 
```

Comment:
Hi @lnu can we move all this to the Dialect? We really need to have one method, which will generate the constraint. Reason for that, is that MS SQL does support "if exist" statement natively. Also the `GetIfExistsDropConstraintEnd` method looks more like a workaround than a solution.